packages/gatsby-plugin-nginx/src/nginx-generator.ts

```diff
@@ -115,23 +111,26 @@ function convertToPath(path: string) {
   return path.replace(/:splat/g, '$1')
 }
 
-function validateRedirect({ toPath }: Redirect): boolean {
+function parseRewrite({
+  toPath,
+  statusCode = 200,
+}: Redirect): 'proxy' | 'rewrite' | 'error_page' {
   try {
     new URL(toPath)
+
+    return 'proxy'
   } catch (ex) {
-    throw new Error(`redirect toPath "${toPath}" must be a valid absolute URL`)
+    // This wasn't a proxy url. It must be an internal redirect
```

Comment:
is this comment necessary?

Reply:
If I leave the `catch` clause without anything inside the linter complains. I thought that adding a comment of why I don't do anything inside this clause might help understanding what's happening in here. I will try to add a better comment in here